lib/OpenQA/BuildResults.pm

```diff
@@ -123,29 +123,34 @@ sub compute_build_results {
     }
 
     # split the statement - give in to perltidy
-    my $search_opts = {
-        select => ['VERSION', 'BUILD', {min => 't_created', -as => 'first_hit'}],
-        as     => [qw(VERSION BUILD first_hit)],
-        # order-by is required here despite it is not relevant when iterating results
-        # because the ordering must be applied before the row limit is applied
-        order_by => {-desc => 'first_hit'},
-        group_by => [qw(VERSION BUILD)]};
-    $search_opts->{rows} = $limit if defined($limit);
-    my $search_filter = {group_id => {in => $group_ids}};
+    my %search_opts = (
+        select   => [qw(VERSION BUILD)],
+        group_by => [qw(VERSION BUILD)],
+        rows     => (defined($limit) && $limit > 400) ? $limit : 400
```

Comment:
> the decision to limit on 400 even if there is a larger limit given

If there is a larger limit given the query isn't limited on 400 (at least this is the intention behind the code).

But this leaves my other question whether it makes sense at all open.